freqtrade/freqai/data_kitchen.py

```diff
@@ -85,9 +85,10 @@ def __init__(
                 config["freqai"]["backtest_period_days"],
             )
 
-        db_url = self.config.get('db_url', None)
-        self.database_path = '' if db_url == 'sqlite://' else str(db_url).split('///')[1]
-        self.trade_database_df: DataFrame = pd.DataFrame()
+        if self.live:
+            db_url = self.config.get('db_url', 'sqlite://')
+            self.database_path = '' if db_url == 'sqlite://' else str(db_url).split('///')[1]
```

Comment:
It's unsafe to assume tripple `/` in the dburl.
That's only valid for sqlite url's.
Using postgres or mysql will get you a keyerror on the `[1]` - as the split won't yield a 2nd element.

Reply:
Apart from that it's also breaking CI - as database_path is no longer assigned - and therefore a "use before definition" error.